airs2icarol/airs2icarol.py

```diff
@@ -203,22 +199,28 @@ def _convert_part(error_log, processed, iterable, root, tagname, culture=None, a
             # record that we have seen this id
             processed.add(record_id)
 
-            # get the translation for the default join string
+            # get the translation fo_airs_type_mappingr the default join string
             joinstr = _(u'; ')
 
             # iterate over mapping definitions
-            for mapping in _airs_type_mapping[tagname]:
-
-                # grab the value for this row of the CSV
-                value = mapping.extractfn(element, joinstr=joinstr, _=_)
-                if value:
-                    value = value.strip()
-
-                # only output the row if we got a value
-                if value:
-                    # agency key, record key, iCarol record type, iCarol Field ID, iCarol Field Name, culture code, value
-                    yield [agencynum or '', key, record_type, unicode(mapping.FieldID), mapping.FieldName, culture, value.replace(u'\n', '\r\n')]
-
+			
+            temp = [agencynum or '', key, record_type, culture]
+            type_mapping = _airs_type_mapping[tagname]
+            for item in _field_order:
+                try:
+                    fn = type_mapping[item]
+                    value = fn(element, joinstr=joinstr, _=_)
+                    if value:
+                        value = value.strip()
+                        temp.append(value)
+                    else:
+                        temp.append('')
+                        #print tagname, item
+                except KeyError:
+                    temp.append('')
```

Comment:
The "except" needs to be as narrowly scoped as possible, by directly following the thing you are trying to catch (e.g. put it after line 211 here so you don't go to the exception from an error on 212 etc.). Use a try-except-else pattern.

You should bo focusing on the setting of *value* and then do temp.append at the end. This should also make it easier to default to an empty string, and also make sure that temp.append is completed each time which is critical to making the rows line up properly.

Try a simplification of code by using
``` py
value = (value or '').strip()
```